solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -73,6 +82,25 @@ fn get_token_mint(denom: &PrefixedDenom) -> Result<Pubkey, TokenTransferError> {
     Ok(Pubkey::find_program_address(&seeds, &crate::ID).0)
 }
 
+/// Removes the destination source and port id and 
+/// returns the hash of full denom.
+pub fn get_hashed_full_denom(denom: &PrefixedDenom) -> CryptoHash {
+    let mut prefixed_denom = denom.clone();
+    let trace_path = prefixed_denom.trace_path.to_string();
+    let mut trace_path = trace_path.split('/');
+    let dest_port_id = trace_path.next();
+    let dest_channel_id = trace_path.next();
+    match (dest_port_id, dest_channel_id) {
+        (Some(port_id), Some(channel_id)) => prefixed_denom.remove_trace_prefix(&TracePrefix::new(
+            PortId::from_str(port_id).unwrap(),
+            ChannelId::from_str(channel_id).unwrap(),
+        )),
+        (_, _) => ()
+        };
+    let full_denom = prefixed_denom.to_string();
+    CryptoHash::digest(full_denom.as_bytes())
+}
```

Comment:
I’d like to see some unit tests for this because it becomes unclear
what this actually dose.  From what I understand, it takes prefix such
as `foo/bar/baz/qux` and then returns hash of `baz/qux`?  Except if
prefix has only one component in which case it returns hashs of the
full prefix.

Also if that’s what this function is doing than using `match_indices`
may be better:

```suggestion
/// Removes the destination source and port id and
/// returns the hash of full denom.
pub fn get_hashed_full_denom(denom: &PrefixedDenom) -> CryptoHash {
    let trace_path = denom.trace_path.to_string();
    let idx = trace_path.match_indices('/').nth(1).map_or(0, |idx| idx + 1);
    CryptoHash::digest(trace_path.as_bytes()[idx..])
}
```

Reply:
Also, shouldn’t this take base denom into account?  Something like

    CryptoHash::digestv(&[trace_path.as_bytes()[idx..], denom.base_denom.as_str().as_bytes()])

Reply:
if you convert the `PrefixedDenom` to string then no need but if you only consider `trace_path` then you need to do it. So as per your suggestion we need to append base denom too.

Reply:
tbh, this is only called during minting and burning in which it should always have a prefix

Reply:
closing since we have removed this.